### ISSUE TITLE
Add material override in render settings

### ISSUE THREAD

**Issue by dictoon** [appleseedhq/appleseed-max#101]

**Comment by usakhelo** [comment]
Is this global material that will be applied to all the objects?
Just an idea - can there be a choice like "override all the materials" and "override only solid materials"? This is very helpfull in architecture where you want do override materials but still want light to get through windows, lights etc.

**Comment by dictoon** [comment]
That's an interesting idea. Another class of materials we could want to exclude is light-emitting ones. It becomes a bit tricky if a material has a sub-material that emits light. I'm not sure how far we need to go here.

**Comment by juancarlosgzrz** [comment]
It could have several options:

[ ] Global Subtitution
[ ] Exclude Lights
[ ] Exclude Glass 

Or to save some UI space, they could be inside a dropdown list to choose from all these options. I recommend that the Override Material have 225 white in diffuse color, no specular and 50%/100%  roughness.

**Comment by usakhelo** [comment]
Yes, lights too. That would be cool. What if it happens in create_material function, so that each material knows should it create itself or a substitute? That way blend, light and glass material just wouldn't care.

**Comment by dictoon** [comment]
@juancarlosgzrz That's pretty much the UI I had in mind. However, the user should be allowed to plug her own material; we shouldn't impose a substitution material.

**Comment by dictoon** [comment]
@usakhelo I think the switch between the scene material and the substitution material will happen much higher in the callstack (because the substitution material will just be a normal material).

**Comment by juancarlosgzrz** [comment]
@dictoon True that

**Comment by juancarlosgzrz** [comment]
Done?
